teammapper-backend/src/map/controllers/maps.gateway.ts

```diff
@@ -176,14 +177,30 @@ export class MapsGateway implements OnGatewayDisconnect {
       return Promise.resolve(false)
 
     const mmpMap: IMmpClientMap = request.map
+
+    // Disconnect all clients temporarily
+    // Emit an event so clients can display a notification
+    this.server.to(mmpMap.uuid).emit('clientNotification', { clientId: client.id, message: 'TOASTS.WARNINGS.MAP_IMPORT_IN_PROGRESS', type: 'warning' })
+
+    const sockets = await this.server.in(request.mapId).fetchSockets()
+    this.server.in(request.mapId).socketsLeave(request.mapId)
+
     await this.mapsService.updateMap(mmpMap)
 
+    // Reconnect clients once map is updated
+    sockets.forEach((socket: RemoteSocket<any, any>) => {
```

Comment:
hope there are better types available than any any? ;-)

Reply:
Unfortunately not, because afaik it returns `RemoteSocket<DefaultEventMaps, any>`, but `socket.io` doesn't export `DefaultEventMaps`. Haven't found anything in the docs regarding `RemoteSocket` either, but I'll take another look.

Reply:
the other question is also if you need to explicitly specify the type as usually it should be inferred from the list of `fetchSockets()`. So maybe just remove the type definition. If the library doesn't specify more its ok but I don't like having `any` in the source code.

Reply:
I changed it so the type is inferred from usage.